SimTracker/SiPixelDigitizer/plugins/SiPixelChargeReweightingAlgorithm.cc

```diff
@@ -562,3 +572,203 @@ void SiPixelChargeReweightingAlgorithm::printCluster(float arr[TXSIZE][TYSIZE])
   }
   std::cout.copyfmt(std::ios(nullptr));
 }
+
+bool SiPixelChargeReweightingAlgorithm::lateSignalReweight(const PixelGeomDetUnit* pixdet,
+                                                           std::vector<PixelDigi>& digis,
+                                                           PixelSimHitAddExtraInfo& loopTempSH,
+                                                           signal_map_type& theNewDigiSignal,
+                                                           const TrackerTopology* tTopo,
+                                                           CLHEP::HepRandomEngine* engine) {
+  uint32_t detID = pixdet->geographicalId().rawId();
+  const PixelTopology* topol = &pixdet->specificTopology();
+
+  if (UseReweighting) {
+    std::cout << " ******************************** " << std::endl;
```

Comment:
No thread-unsafe `cout` in production code.

Reply:
I think this can be a `LogError` at least.
If we want really to enforce this should not happen it should throw.